src/Qir/Execution/Tools/Executable/QirFullStateExecutable.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Quantum.Qir.Tools.Driver;
+using Microsoft.Quantum.Qir.Utility;
+
+namespace Microsoft.Quantum.Qir.Tools.Executable
+{
+    /// <summary>
+    /// Class to create and run QIR-based executables that use the full-state simulator.
+    /// </summary>
+    public class QirFullStateExecutable : QirExecutable
+    {
+        internal QirFullStateExecutable(FileInfo executableFile, byte[] qirBytecode, ILogger logger, IQirDriverGenerator driverGenerator, IQirExecutableGenerator executableGenerator, IQuantumExecutableRunner runner)
+            : base(executableFile, qirBytecode, logger, driverGenerator, executableGenerator, runner)
+        {
+
+        }
+
+        public QirFullStateExecutable(FileInfo executableFile, byte[] qirBytecode, ILogger logger)
```

Comment:
Can you please make the `logger` argument optional? In the context of the QIR controller having a logger makes sense but for the extended purposes of this library it makes sense to provide an option that does not require a logger.